lightning/src/ln/channel.rs

```diff
@@ -4020,6 +4013,15 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 	}
 }
 
+fn is_unsupported_shutdown_script(their_features: &InitFeatures, scriptpubkey: &Script, is_outbound: bool) -> bool {
+	// BOLT 2 says we must only send a scriptpubkey of certain standard forms,
+	// which for a a BIP-141-compliant witness program is at max 42 bytes in length.
+	// So don't let the remote peer feed us some super fee-heavy script.
+	let is_script_too_long = is_outbound && scriptpubkey.len() > 42;
+	let not_valid_witness = !their_features.supports_shutdown_anysegwit() && scriptpubkey.is_witness_program();
+	return is_script_too_long || (not_valid_witness && !scriptpubkey.is_p2pkh() && !scriptpubkey.is_p2sh() && !scriptpubkey.is_v0_p2wpkh() && !scriptpubkey.is_v0_p2wsh())
```

Comment:
I don't think we need to check script length at all. P2PKH, P2SH and segwit scriptpubkeys by defintion MUST be inferior to 42 bytes.

For `not_valid_witness`, I think you need to pass our features support to check that our node is supporting `shutdown_anysegwit`.

Reply:
The `is_script_too_long` thing is also applied for anysegwit, so I think its fine?

As for checkout our own features - we always use `Features::supported()` so there's nothing to check against. We can revisit if we build a way for users to configure different features themselves.

Reply:
Ah! `is_witness_program` enforces size already https://docs.rs/bitcoin/0.26.0/src/bitcoin/blockdata/script.rs.html#366. I don't mind (kinda like) enforcing it double, but could be commented to note that.

Reply:
Hmmmmm, I would keep the check because the check inside `is_witness_program` should only be done if they support shutdown any segwit (I'll comment on the condition in the other comment).